src/rebar3_hex.erl

```diff
@@ -1,13 +1,19 @@
 -module(rebar3_hex).
 
 -export([ init/1
+        , get_args/1
         , gather_opts/2
         , get_required/2
         , task_args/1
+        , task_state/1
         , repo_opt/0
         , help_opt/0
         ]).
 
+-type task() :: #{args := map(), repo := map(), state := rebar_state:t()}.
```

Comment:
I opted not to call this `task_state` or `state` as in `rebar3_hex:state/0`. I also opted not to name the key `state`, `rebar_state`. This is all internal, no one should be using this as a lib, and thus it's just implementation details. I think it would be confusing to have too many things called `*state*`. Likewise, `state` in a rebar3 plugin is synonymous with rebar3 state to me. Open to suggestion if there's strong feelings. Of course, like a lot of things we can always change it again while busting up the interface and refactoring further.